### DISCUSSION TITLE
Libraries for cross-compilation to x86_64-pc-windows-gnu

### DISCUSSION THREAD

**Discussion by DwarfsLord** [cross-rs/cross#1056] [Q&A] [answered]
Hello,
newbie here, so I probably missed something obvious.

I am trying to cross-compile from linux (arch) to "x86_64-pc-windows".
My project relies on diesel (database ORM) which in turn depends on libmysqlclient.

Everyting compiles fine for "x86_64-unknown-linux-gnu", but I am getting following error when I do:

cross build --release --target x86_64-pc-windows-gnu

> linking with `x86_64-w64-mingw32-gcc` failed: exit status: 1
> [...]
> note: /usr/bin/x86_64-w64-mingw32-ld: cannot find -lmysqlclient
>           collect2: error: ld returned 1 exit status

I tried to have a Dockerfile with this:

> FROM ghcr.io/cross-rs/x86_64-pc-windows-gnu:latest
> 
> RUN dpkg --add-architecture x86-64 && \
>     apt-get update && \
>     apt-get install --assume-yes libmysqlclient-dev

but still get the same error.

So, how do I get these libraries?
Or am I misunderstanding my problem?

Thank you for any help!

**Comment by Alexhuszagh** [comment, marked as answer]
The target you are building for, `x86_64-pc-windows-gnu`, is using GCC for a Windows target, but the host OS is x86_64 Linux GNU. So, installing `libmysqlclient-dev` via `apt` installs it for `x86_64-unknown-linux-gnu`. If you would like to use a crate depending on your library, you'll have to build it from source or use a package manager that has it built. Some packages, like `libz-mingw-w64-dev`, are available from official repositories, but many others aren't.

A few tools and frameworks help for this. For example, `vcpkg` could let you use it as part of your build step, or you can download it from source and compile it with our toolchain. You can also use [MXE](https://github.com/mxe/mxe) (M Cross-Environment).
